src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -95,6 +95,244 @@ public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSy
         public static FunctionPointerMethodSymbol CreateFromMetadata(CallingConvention callingConvention, ImmutableArray<ParamInfo<TypeSymbol>> retAndParamTypes)
             => new FunctionPointerMethodSymbol(callingConvention, retAndParamTypes);
 
+        public FunctionPointerMethodSymbol SubstiteParameterSymbols(
+            TypeWithAnnotations substitutedReturnType,
+            ImmutableArray<TypeWithAnnotations> substitutedParameterTypes,
+            ImmutableArray<CustomModifier> refCustomModifiers,
+            ImmutableArray<ImmutableArray<CustomModifier>> paramRefCustomModifiers)
+            => new FunctionPointerMethodSymbol(
+                this.CallingConvention,
+                this.RefKind,
+                substitutedReturnType,
+                refCustomModifiers.IsDefault ? this.RefCustomModifiers : refCustomModifiers,
+                this.Parameters,
+                substitutedParameterTypes,
+                paramRefCustomModifiers);
+
+        internal FunctionPointerMethodSymbol MergeEquivalentTypes(FunctionPointerMethodSymbol signature, VarianceKind variance)
```

Comment:
>MergeEquivalentTypes [](start = 45, length = 20)

Do we have tests that make the fixes around ref-ness observable?

Reply:
I'll add an extension to do this for ref as well, I have one for nonref

---
In reply to: [406970678](https://github.com/dotnet/roslyn/pull/43041#discussion_r406970678) [](ancestors = 406970678)

Reply:
Added `BestTypeForConditional_NestedParameterRef`

---
In reply to: [406983797](https://github.com/dotnet/roslyn/pull/43041#discussion_r406983797) [](ancestors = 406983797,406970678)